lib/compiler/include/compiler/unity_algorithm.h

```diff
@@ -1,24 +1,77 @@
 #ifndef _FLEXFLOW_FFC_UNITY_ALGORITHM_H
 #define _FLEXFLOW_FFC_UNITY_ALGORITHM_H
 
-#include "utils/graph.h"
-#include "op-meta/operator_attrs.h"
+#include "op-attrs/operator_attrs.h"
 #include "pcg/machine_view.h"
+#include "pcg/machine_specification.h"
+#include "utils/graph.h"
+#include "compiler.h"
+#include "cost_estimate.h"
 
 namespace FlexFlow {
 
-/* std::unordered_map<MultiDiEdge, ParallelTensorShape> infer_tensor_shapes(ParallelComputationGraph const &); */
+using OptimizerComputationGraph =
+    NodeLabelledMultiDiGraph<ComputationGraphAttrs>;
+using OptimizerPCG =
+    LabelledMultiDiGraph<PCGOperatorAttrs, ParallelTensorShape>;
+
+using SubParallelComputationGraph =
+    LabelledOpenMultiDiGraph<PCGOperatorAttrs,
+                             ParallelTensorShape,
+                             MachineView>;
+
+struct Substitution {
+};
+
+struct Strategy {
```

Comment:
I've move `Strategy` out into its own `.h`/`.cc` files to help the reader not get lost in big files of lots of definitions